app/plugin_key_value_store.go

```diff
@@ -68,6 +68,26 @@ func (a *App) CompareAndSetPluginKey(pluginId string, key string, oldValue, newV
 	return updated, nil
 }
 
+func (a *App) CompareAndDeletePluginKey(pluginId string, key string, oldValue []byte) (bool, *model.AppError) {
+	kv := &model.PluginKeyValue{
+		PluginId: pluginId,
+		Key:      key,
+	}
+
+	deleted, err := a.Srv.Store.Plugin().CompareAndDelete(kv, oldValue)
+	if err != nil {
+		mlog.Error("Failed to compare and delete plugin key value", mlog.String("plugin_id", pluginId), mlog.String("key", key), mlog.Err(err))
+		return deleted, err
```

Comment:
Should we do `return nil, err` here instead?

Reply:
it should be `false, err`, right? I totally agree :)

Reply:
Yes, I meant `return false, err` 🤦‍♀️

Reply:
the benefit of returning `deleted, err` here is then the logic that determines what `deleted` is remains in one place, i.e. inside `a.Srv.Store.Plugin().CompareAndDelete(kv, oldValue)`. If we decide to be explicit about this then maybe we should update `return updated, err` to `return false, err` to be consistent.

[0/5](https://docs.mattermost.com/process/training.html#id8) on this.

Reply:
👍 	for consistency with the existing `return updated, err` pattern.

Reply:
thanks @ali-farooq0, I agree that we should be consistent if we go the `return false, err` logic (which personally I prefer). But you're talking about [L60](https://github.com/mattermost/mattermost-server/pull/11804/files#diff-afd8e004133130ca459a180c0ed896fdR60) (not [L68](https://github.com/mattermost/mattermost-server/pull/11804/files#diff-afd8e004133130ca459a180c0ed896fdL68)), right?

Reply:
seeing that there seems to be agreement that `updated, err` is better than `false, err` I went with that approach on https://github.com/mattermost/mattermost-server/pull/11804/commits/9ef6d9a45dc35fdd534deac8fa718314a1eed1ca